lib/record_loader/asset_shape_loader.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+# RecordLoader handles automatic population and updating of database records
+# across different environments
+# @see https://rubydoc.info/github/sanger/record_loader/
+module RecordLoader
+  class AssetShapeLoader < ApplicationRecordLoader
+    config_folder 'asset_shapes'
+
+    # Creates an AssetShape record with the given name and options. If a record
+    # with the same name already exists, it is skipped. If not, a new one is
+    # created. For each size specified in the options, Map records are generated
+    # up to that size, unless they already exist.
+    #
+    # @param name [String] the name of the AssetShape record
+    # @param options [Hash] the options to be used for creating the record
+    # @param options [String] :name If provided, this will override the 'name' parameter
+    # @option options [Integer] :horizontal_ratio The horizontal ratio of the plate
+    # @option options [Integer] :vertical_ratio The vertical ratio of the plate
+    # @option options [String] :description_strategy The strategy for describing the plate
+    # @option options [Array<Integer>] :sizes The sizes of the plates to generate Maps for
+    def create_or_update!(name, options)
+      config = { name: name }.merge(options.symbolize_keys)
+      config[:description_strategy] = config[:description_strategy].delete_prefix('Map::')
```

Comment:
Where does this 'Map::' prefix come from? I don't see it in config/default_records/asset_shapes/default_records.yml ?

Reply:
Now I have added the following comment just before the line to explain.

# PlateMapGeneration expects a non-namespaced constant for
# description_strategy. It adds "Map::" prefix to refer to a nested
# module in the Map class. We remove this prefix in case it is given
# in the records file.

I will push the change shortly.

Reply:
Ah so you're future proofing in case someone follows a different convention to you when using record loader?

Reply:
I have reused the PlateMapGeneration class and kept the existing behaviour. I did not want to change how it works. It is also called from the spec_helper before-suit-hook to initialise the test database, and from seeds to initialise the development database. The name-spaced name is written in the asset_shapes table's description_strategy column.